ICSharpCode.Decompiler/IL/Transforms/TransformArrayInitializers.cs

```diff
@@ -187,6 +221,27 @@ static bool MatchSpanTCtorWithPointerAndSize(NewObj newObj, StatementTransformCo
 			return true;
 		}
 
+		static bool MatchRuntimeHelpersCreateSpan(Call inst, StatementTransformContext context, out IType elementType, out FieldDefinition field)
+		{
+			field = default;
+			elementType = null;
+			IType type = inst.Method.DeclaringType;
+			if (type.Namespace != "System.Runtime.CompilerServices" || type.Name != "RuntimeHelpers" || type.TypeParameterCount != 0)
+				return false;
+			if (inst.Arguments.Count != 1)
+				return false;
+			IMethod method = inst.Method;
+			if (method.Name != "CreateSpan" || method.TypeArguments.Count != 1)
+				return false;
```

Comment:
Should I be checking the full method signature? I noticed that the other handler doesn't check argument types.

Reply:
checking more is usually better than checking less... note that the code in that file is very old and thus is not using some of the helpers and is not following the style of later transform implementations

Reply:
Addendum: for methods we usually check the name, the number of parameters and then validate all the arguments, so checking the parameter types is usually not necessary.